specification/liftrqumulo/Qumulo.Storage.Management/LiftrBase.Storage/main.tsp

```diff
@@ -0,0 +1,137 @@
+import "./../LiftrBase/main.tsp";
+
+import "@typespec/openapi";
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+
+using Azure.ResourceManager;
+using LiftrBase;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using TypeSpec.Reflection;
+using Azure.ResourceManager.Foundations;
+
+@versioned(LiftrBase.Storage.Versions)
+@armLibraryNamespace
+namespace LiftrBase.Storage;
+
+@doc("Supported versions for LiftrBase.Storage resource model")
+enum Versions {
+  @doc("Dependent on Azure.ResourceManager.Versions.v1_0_Preview_1 and LiftrBase.Versions.v1_preview")
+  @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+  @useDependency(LiftrBase.Versions.v2_preview)
+  @armCommonTypesVersion(Azure.ResourceManager.CommonTypes.Versions.v3)
```

Comment:
Same here, please update all the Versions enum

Reply:
These version changes are coming because of typespec onboarding. Let me explain.

There are other models (apart from Managed Identity) that we are using from v3. Example: https://github.com/Azure/azure-rest-api-specs/blob/6d47e106e5dbe91ee5d124785fe1779fc170cac9/specification/liftrqumulo/resource-manager/Qumulo.Storage/stable/2022-10-12/Qumulo.Storage.json#L409

Since the count of types being referred were more for v3 than for v4, we decided to use the armCommonTypesVersion as v3. Because of a common decorator, it started using v3 even for Managed Identity models.
In reality, there shouldn't be any issue as v3 and v4 are identical for Managed identity and we want to maintain v3 for all other models. Hope this explains.